SteamKit2/SteamKit2/Steam/Discovery/SmartCMServerList.cs

```diff
@@ -264,7 +233,8 @@ private IPEndPoint GetNextServerCandidateInternal()
                 ResetOldScores();
 
                 var serverInfo = servers
-                    .Select( (s, index) => new { EndPoint = s.EndPoint, IsBad = s.LastBadConnectionDateTimeUtc.HasValue, Index = index } )
+                    .Where( s => (s.Record.ProtocolTypes & supportedProtocolTypes) > 0 )
```

Comment:
Is this a hot path? An extension method is probably fine, not sure if it makes much of a perf difference between that and `HasFlag`, and if it's worth it. :)